ADR/0017-posture-keeper.md

```diff
@@ -0,0 +1,59 @@
+# 17. Posture Keeper
+
+Date: 2023-02-18
+
+## Status
+
+Accepted
+
+## Context
+Posture keeper is a REST-based web service that allows clients to store and retrieve SAST analysis data for a specific git repository and commit SHA. The service is designed to be scalable, reliable, and secure, and uses MongoDB as its underlying data store. The service is implemented using Quarkus. By utilizing this service StoneSoup is going to be able to provides experiences that reflects the current state of security and/or quality posture of the applications.
```

Comment:
The scope is mixed here. On the one hand, it is specifically about SAST. On the other hand, the text suggests that it supports security and/or quality records more generally. Which is it?

Reply:
Security and quality can still be derived from static scans. Are you trying to ask if we are having an `and` or an `or` and asking to not have both?

I imagine that the language is in there because it depends on the scans that were run against a snapshot at whatever point in time -- whether they were primarily security or quality related.